src/core/src/FabricMaterial.cpp

```diff
@@ -583,7 +586,27 @@ void FabricMaterial::initializeNodes() {
 void FabricMaterial::initializeDefaultMaterial() {
     auto srw = UsdUtil::getFabricStageReaderWriter();
 
-    const auto imageryLayerCount = getImageryLayerCount(_materialDefinition);
+    int overlayImageryLayerCount = 0, clippingImageryLayerCount = 0;
+    uint64_t layerNum = 0;
+    std::vector<uint64_t> overlayImageryLayerIndices, clippingImageryLayerIndices;
+
+    for (auto pipeType : _materialDefinition.getImageryOverlayRenderPipes()) {
+        switch (pipeType) {
+            case OverlayRenderPipe::OVERLAY:
+                overlayImageryLayerIndices.push_back(layerNum);
+                overlayImageryLayerCount++;
+                break;
+            case OverlayRenderPipe::CLIPPING:
+                clippingImageryLayerIndices.push_back(layerNum);
+                clippingImageryLayerCount++;
+                break;
+        }
+        layerNum++;
+        if (layerNum == MAX_IMAGERY_LAYERS_COUNT) {
```

Comment:
There should be a separate `layerNum` for each render pipe.

The limitation has to do with how many slots there are available for `cesium_internal_imagery_layer_resolver` and `cesium_internal_polygon_imagery_layer_resolver`, not combined.

Reply:
Ah, that's right. The input arrays for the resolver have to be hard-coded because MDL doesn't have dynamic arrays. Changed.